ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -479,19 +494,32 @@ private PropertiesList listProperties(
   private QueryDescriptionEntity explain(
       final PreparedStatement<Explain> statement,
       final Map<String, Object> propertyOverrides
+  ) {
+    return explain(statement, propertyOverrides, ksqlConfig, ksqlEngine);
+  }
+
+  private static QueryDescriptionEntity explain(
+      final PreparedStatement<Explain> statement,
+      final Map<String, Object> propertyOverrides,
+      final KsqlConfig ksqlConfig,
+      final KsqlExecutionContext executionContext
   ) {
     final String queryId = statement.getStatement().getQueryId();
 
-    final QueryDescription queryDescription = queryId == null
-        ? explainStatement(
-        statement.getStatement().getStatement(),
-        statement.getStatementText().substring("EXPLAIN ".length()),
-        ksqlEngine,
-        ksqlConfig,
-        propertyOverrides)
-        : explainQuery(queryId, ksqlEngine);
+    try {
+      final QueryDescription queryDescription = queryId == null
+          ? explainStatement(
+          statement.getStatement().getStatement(),
+          statement.getStatementText().substring("EXPLAIN ".length()),
+          executionContext,
+          ksqlConfig,
+          propertyOverrides)
+          : explainQuery(queryId, executionContext);
 
-    return new QueryDescriptionEntity(statement.getStatementText(), queryDescription);
+      return new QueryDescriptionEntity(statement.getStatementText(), queryDescription);
+    } catch (final KsqlException e) {
+      throw new KsqlStatementException(e.getMessage(), statement.getStatementText(), e);
```

Comment:
While in here I've improved the exception type being thrown for many statements to ensure the problem statement is returned.